Run Method  추가 및 Shell Interface 정리

flowinto10 opened the pull request:
Shell Program에 Run method를 추가하고, 불필요한 인터페이스를 정리하였습니다.

flowinto10 commented on TeamBest_SSD_Shell/SSD_Shell.cpp:
```diff
@@ -1,7 +1,32 @@
+    void Run(void) {
+        int loopCount = 0;
+        std::string line;
+        while (loopCount < 5) {
```
Shell program 은 특성상 사용자의 커맨드를 지속적으로 입력받기 위해, 
무한 loop을 돌아야 하나, 그럴경우 TC가 종료되지 않아 (아직 Exit 구현전) 이를 위해 loop 횟수를 제한합니다.
추후 exit 구현후 while (true)로 변경 예정입니다.

flowinto10 commented on TeamBest_SSD_Shell/main.cpp:
```diff
@@ -8,6 +8,11 @@ TEST(ShellTS, CraeteShellInstance) {
 
+TEST(ShellTS, CheckRunMethod) {
+    SSDShell* ssdShell = new SSDShell();
+    EXPECT_NO_THROW(ssdShell->Run());
```
run 함수 구현을 위한 TC입니다.

flowinto10 commented on TeamBest_SSD_Shell/ISSD_Shell.h:
Shell은 현재 interface가 필요하지 않아 보여, 제거 하였습니다.

flowinto10 commented on TeamBest_SSD_Shell/SSD_Shell.h:
인터페이스 이름들을 좀 더 직관적으로 정리하였습니다.

flowinto10 commented on TeamBest_SSD_Shell/test_ssdshell1.cpp:
파일 Conflict을 피하기 위해, TC파일 분리

R1NRINRIN reviewed (approved):
TODO까지 한 눈에 볼 수 있어 좋습니다. 고생하셨습니다.

youngchul-hwang reviewed (approved):
PR에 추가적인 설명을 자세히 적어주셔서
코드 읽고 파악하기가 좋았네요.
수고하셨습니다.
충돌이 있는 것 같은데, 그것만 해결하고 merge하면 될 것 같습니다.

Ellen99999999 commented on TeamBest_SSD_Shell/SSD_Shell.h:
```diff
@@ -0,0 +1,25 @@
+
+	void Run(void);
+	void PrintHelp() {};
+	std::string ReadSSDOutputFile(int address) { return 0x0; }
```
생각해보니... output format 은 string 인데, return 0 보다 return "" 이 낫지 않을까요?

Ellen99999999 replied:
제가 SHA-1: a7f5cdc17c2 에 푸시 한거랑 함수명이 달라서 그 부분 같이 업데이트 되어야 빌드에 문제가 없을 거 같습니다.

Ellen99999999 commented on TeamBest_SSD_Shell/SSD_Shell.cpp:
```diff
@@ -1,10 +1,22 @@
+void SSDShell::Run(void) {
+    int loopCount = 0;
+    std::string line;
+    while (loopCount < 5) {
```
loopCount 5 는 임시로 만드신거죠?

flowinto10 committed:
conflict resolve

flowinto10 committed:
[Feature] Add Run Interface
1. 변경점
- Add run interface
- shell interface 변경
2.검증

[X] Build

[X] Regression Test Pass

[X] 해당 기능 TC
